demo/src/constrained_planning_demo.cpp

```diff
@@ -0,0 +1,279 @@
+
+  // In the second problem we plan with the end-effector constrained to a plane.
+  // We need to create a pose goal that lies in this plane.
+  // The plane is tilted by 45 degrees, so moving an equal amount in the y and z direction should be ok.
+  // Any goal or start state should also satisfy the path constraints.
+  target_pose = get_relative_pose(0.0, 0.3, -0.3);
+
+  // We create a plane perpendicular to the y-axis and tilt it by 45 degrees
+
+  // Solving the problem using equality constraints is a bit more complicated. (Or should I say, hacky?)
+  // We need to tell the planner explicitly that we want to use equality constraints for the small dimensions.
+  // This is achieved by setting the name of the constraint to :code:`"use_equality_constraints"`.
+  // In addition, any dimension of the box below a threshold of :code:`0.001` will be considered an equality constraint.
+  // However, if we make it too small, the box will be thinner that the tolerance used by OMPL to evaluate constraints
+  // (:code:`1e-4` by default). MoveIt will use the stricter tolerance (the box width) to check the constraints, and
+  // many states will appear invalid. That's where the magic number :code:`0.0005` comes from, it is between
+  // :code:`0.00001` and :code:`0.001`.
+  moveit_msgs::msg::PositionConstraint plane_constraint;
+  plane_constraint.header.frame_id = move_group_interface.getPoseReferenceFrame();
+  plane_constraint.link_name = move_group_interface.getEndEffectorLink();
+  shape_msgs::msg::SolidPrimitive plane;
+  plane.type = shape_msgs::msg::SolidPrimitive::BOX;
+  plane.dimensions = { 1.0, 0.0005, 1.0 };
+  plane_constraint.constraint_region.primitives.emplace_back(plane);
+
+  geometry_msgs::msg::Pose plane_pose;
+  plane_pose.position.x = current_pose.pose.position.x;
+  plane_pose.position.y = current_pose.pose.position.y;
+  plane_pose.position.z = current_pose.pose.position.z;
+  plane_pose.orientation.x = sin(M_PI_4 / 2);
+  plane_pose.orientation.y = 0.0;
+  plane_pose.orientation.z = 0.0;
+  plane_pose.orientation.w = cos(M_PI_4 / 2);
+  plane_constraint.constraint_region.primitive_poses.emplace_back(plane_pose);
+  plane_constraint.weight = 1.0;
+
+  // Visualize the constraint
+  auto d = sqrt(pow(target_pose.pose.position.y, 2) + pow(target_pose.pose.position.z, 2));
+
+  Eigen::Vector3d normal(0, 1, 1);
+  moveit_visual_tools.publishNormalAndDistancePlane(normal, d, rviz_visual_tools::TRANSLUCENT_DARK);
+  moveit_visual_tools.trigger();
+
+  moveit_msgs::msg::Constraints plane_constraints;
+  plane_constraints.position_constraints.emplace_back(plane_constraint);
+  plane_constraints.name = "use_equality_constraints";
+  move_group_interface.setPathConstraints(plane_constraints);
+
+  // And again, configure and solve the planning problem
+  move_group_interface.setPoseTarget(target_pose);
+  move_group_interface.setPlanningTime(10.0);
+
+  // Use KTOpt
+  move_group_interface.setPlanningPipelineId("drake_ktopt");
```

Comment:
Once KTOpt is loaded you should be able to configure it like this or with a function like this from the move_group interface API